File: src/TeamCloud.Model/Data/Core/Provider.cs
/**
 *  Copyright (c) Microsoft Corporation.
 *  Licensed under the MIT License.
 */

using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using TeamCloud.Serialization;

namespace TeamCloud.Model.Data.Core
{
    [JsonObject(NamingStrategyType = typeof(TeamCloudNamingStrategy))]
    public interface IProvider : IProperties
    {
        string Id { get; set; }

        string AuthCode { get; set; }

        Guid? PrincipalId { get; set; }

        AzureResourceGroup ResourceGroup { get; set; }

        IList<string> Events { get; set; }

        DateTime? Registered { get; set; }

        ProviderCommandMode CommandMode { get; set; }
    }
}

File: web/teamcloud/src/model/Provider.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { IIdentifiable, IProperties, AzureResourceGroup } from "./index";

export interface Provider extends IIdentifiable, IProperties {
    id: string;
    url: string;
    // authCode: string;
    principalId: string;
    version: string;
    resourceGroup: AzureResourceGroup;
    events?: string[];
    properties?: Record<string, string>;
    registered?: string;
    commandMode: ProviderCommandMode;
}


export enum ProviderCommandMode {
    Simple = 'Simple',
    Extended = 'Extended'
}
